node-data/src/events/transactions.rs

```diff
@@ -0,0 +1,168 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use super::*;
+use crate::ledger::{Hash, SpentTransaction, Transaction};
+
+#[derive(Clone, Debug)]
+pub enum TransactionEvent<'t> {
+    Removed(Hash),
+    Included(&'t Transaction),
+    Executed(&'t SpentTransaction),
+}
+
+impl EventSource for TransactionEvent<'_> {
+    const COMPONENT: &'static str = "transactions";
+
+    fn topic(&self) -> &'static str {
+        match self {
+            Self::Removed(_) => "removed",
+            Self::Executed(_) => "executed",
+            Self::Included(_) => "included",
+        }
+    }
+    fn data(&self) -> Option<serde_json::Value> {
+        match self {
+            Self::Removed(_) => None,
+            Self::Executed(t) => Some(t.to_json()),
+            Self::Included(t) => Some(t.to_json()),
+        }
+    }
+    fn entity(&self) -> String {
+        let hash = match self {
+            Self::Removed(hash) => *hash,
+            Self::Executed(tx) => tx.inner.hash(),
+            Self::Included(tx) => tx.hash(),
```

Comment:
Shouldn't we use `id()` here like we do for `Removed`?

Reply:
I don't think so

Any "new" transaction (included or executed) must be in the full form (because the client is unaware)

"removed" just need an "id" instead, because it's meant to already be known by the client